compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2343,6 +2343,19 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def unapply(x: NoPrefix): true = true
     end NoPrefix
 
+    type FlexibleType = dotc.core.Types.FlexibleType
+
+    object FlexibleTypeTypeTest extends TypeTest[TypeRepr, FlexibleType]:
+      def unapply(x: TypeRepr): Option[FlexibleType & x.type] = x match
+        case x: (Types.FlexibleType & x.type) => Some(x)
+        case _ => None
+    end FlexibleTypeTypeTest
+
+    object FlexibleType extends FlexibleTypeModule:
+      def apply(tp: TypeRepr): FlexibleType = Types.FlexibleType(tp)
```

Comment:
I noticed that `Types.FlexibleType` is a case class with 2 fields but here we only have one. Why is this difference ok? Is this aligned with the representation in TASTy?

Reply:
The first field `lo` is just a nullable version of `hi` computed internally. Hence, we only store the `hi` in TASTy as well.